2016-2017_Starstruck/Module Prototypes/src/LCD.c

```diff
@@ -1,36 +1,125 @@
 #include <main.h>
+#include <string.h>
 
-MenuItem *lcdmInit() {
-  char title[] = "<    Select    >";
-  MenuItem root = {.name = title,
-                   .previous = malloc(sizeof(MenuItem *)),
-                   .next = malloc(sizeof(MenuItem *)),
-                   .select = malloc(sizeof(MenuItem *)),
-                   .action = malloc(sizeof(Function))
-                  };
-  return &root;
+MenuItem *current;
+FILE *port;
+
+void pulseMotor(unsigned char port);
+
+MenuItem *lcdmInit(FILE *lcdPort) {
+  port = lcdPort;
+  lcdInit(lcdPort);
+  lcdClear(lcdPort);
+  lcdSetBacklight(lcdPort, true);
+  lcdSetText(lcdPort, 1, "~~750Evolution~~");
+  lcdSetText(lcdPort, 2, "Initializing...");
+  printf("LCD Initialized. Creating root menu...\n\r");
+  delay(300);
+  // char title[] = "<    Select    >";
+  MenuItem *root = malloc(sizeof(MenuItem *));
+  root = lcdmCreateItem("<   Select     >");
+  (*root).previous = root;
+  (*root).next = root;
+  printf("Menu created. Setting active menu...\n\r");
+  delay(200);
+  current = root;
+  printf("LCD Menu Initialized\n\r");
+  delay(200);
+  return root;
 }
 
 void lcdmAddDefaults(MenuItem *root) {
-  //TODO implement
+  printf("Adding default LCDM features...\n\r");
+  delay(100);
+  printf("Creating Manual Test menu...\n\r");
+  delay(100);
+  MenuItem *manualTest = malloc(sizeof(MenuItem *)),
+           *exitManualTest = malloc(sizeof(MenuItem *));
+  manualTest = lcdmCreateItem("< Manual Test  >");
+  exitManualTest = lcdmCreateItem("<     Back     >");
+  delay(300);
+  (*exitManualTest).previous = exitManualTest;
+  (*exitManualTest).next = exitManualTest;
+  (*exitManualTest).select = manualTest;
+  printf("Adding Manual Test submenus...\n\r");
+  delay(800);
+  char *title;
+  for (int i = 1; i<=10; i++) {
+    title = malloc(sizeof(char)*16);
+    sprintf(title, "<   Motor %d  >",i);
+    delay(100);
+    MenuItem *motor = malloc(sizeof(MenuItem *));
+    motor = lcdmCreateItem(title);
+    printf("%s", (*motor).name);
+
+    (*motor).action = &pulseMotor;

```

Comment:
so this makes a pointer to the function pulseMotor? and a pointer to the MenuItem motor?